include/nbl/asset/interchange/IAssetLoader.h

```diff
@@ -301,6 +337,50 @@ class IAssetLoader : public virtual core::IReferenceCounted
 	{
 		bundle.setAsset(offset,std::move(_asset));
 	}
+
+private:
+	static IAssetLoader::SAssetLoadParams getFindParams(IAssetLoader::SAssetLoadParams _params)
+	{
+		_params.cacheFlags = static_cast<E_CACHING_FLAGS>(_params.cacheFlags & ECF_DONT_CACHE_REFERENCES);
+		return _params;
+	}
+	SAssetBundle interm_getAssetInHierarchy_find(IAssetManager* _mgr, io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel, IAssetLoader::IAssetLoaderOverride* _override)
+	{
+		return interm_getAssetInHierarchy(_mgr, _file, _supposedFilename, getFindParams(_params), _hierarchyLevel);
+	}
+	SAssetBundle interm_getAssetInHierarchy_find(IAssetManager* _mgr, const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel, IAssetLoader::IAssetLoaderOverride* _override)
+	{
+		return interm_getAssetInHierarchy(_mgr, _filename, getFindParams(_params), _hierarchyLevel, _override);
+	}
+	SAssetBundle interm_getAssetInHierarchy_find(IAssetManager* _mgr, io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel)
+	{
+		return interm_getAssetInHierarchy(_mgr, _file, _supposedFilename, getFindParams(_params), _hierarchyLevel);
+	}
+	SAssetBundle interm_getAssetInHierarchy_find(IAssetManager* _mgr, const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel)
+	{
+		return interm_getAssetInHierarchy(_mgr, _filename, getFindParams(_params), _hierarchyLevel);
+	}
+	static IAssetLoader::SAssetLoadParams getReloadParams(IAssetLoader::SAssetLoadParams _params, uint32_t _restoreLevels, uint32_t _hierLevel)
+	{
+		_params.cacheFlags = ECF_DUPLICATE_UNTIL_LEVEL(_restoreLevels + _hierLevel);
```

Comment:
you should only overwrite the flags between `[_hierLevel,_hierLevel+_restoreLevels]`

Reply:
hm, right, i mean flags for levels `[0, _hierLevel)` doesnt matter since nobody will ever look at them, but i shouldnt zero-out levels `[_hierLevel+_restoreLevels+1, INF)`

Reply:
hm, actually i think i should because if original flags said to not look into cache for levels `>_hierLevel+_restoreLevels`, then those levels will be pointlessly loaded again. So flags for those levels should be 0 (look into cache and put into cache if loaded; or at least just look into cache -- caching/not caching flags doesnt really matter while reloading since we can be sure they were alerady loaded at least once so they'll always be found)

Reply:
> 
> 
> hm, right, i mean flags for levels `[0, _hierLevel)` doesnt matter since nobody will ever look at them, but i shouldnt zero-out levels `[_hierLevel+_restoreLevels+1, INF)`

but for the sake of formalism you shouldn't overwrite the range `[0, _hierLevel)`